nemo/collections/common/tokenizers/huggingface/auto_tokenizer.py

```diff
@@ -106,38 +106,49 @@ def __init__(
-        if self.tokenizer.sep_token is None:
-            if self.tokenizer.eos_token:
-                special_tokens_dict["sep_token"] = self.tokenizer.eos_token
-            else:
-                special_tokens_dict["sep_token"] = sep_token
-        if self.tokenizer.mask_token is None:
+        if mask_token is not None:
             special_tokens_dict["mask_token"] = mask_token
-        if self.tokenizer.bos_token is None:
-            if self.tokenizer.cls_token:
-                special_tokens_dict["bos_token"] = self.tokenizer.cls_token
-            else:
-                special_tokens_dict["bos_token"] = bos_token
-        if self.tokenizer.eos_token is None:
-            if self.tokenizer.sep_token:
-                special_tokens_dict["eos_token"] = self.tokenizer.sep_token
-            else:
-                special_tokens_dict["eos_token"] = eos_token
-        if self.tokenizer.pad_token is None:
+        if pad_token is not None:
             special_tokens_dict["pad_token"] = pad_token
-        if self.tokenizer.cls_token is None:
-            if self.tokenizer.bos_token:
-                special_tokens_dict["cls_token"] = self.tokenizer.bos_token
-            else:
-                special_tokens_dict["cls_token"] = cls_token
 
-        logging.info(f'Adding special tokens to the tokenizer: {special_tokens_dict}')
+        if sep_token is not None:
+            special_tokens_dict["sep_token"] = sep_token
+        elif self.tokenizer.sep_token is None and self.tokenizer.eos_token:
+            special_tokens_dict["sep_token"] = self.tokenizer.eos_token
+        if bos_token is not None:
+            special_tokens_dict["bos_token"] = bos_token
+        elif self.tokenizer.bos_token is None and self.tokenizer.cls_token:
+            special_tokens_dict["bos_token"] = self.tokenizer.cls_token
+        if eos_token is not None:
+            special_tokens_dict["eos_token"] = eos_token
+        elif self.tokenizer.eos_token is None and self.tokenizer.sep_token:
+            special_tokens_dict["eos_token"] = self.tokenizer.sep_token
+        if cls_token is not None:
+            special_tokens_dict["cls_token"] = cls_token
+        elif self.tokenizer.cls_token is None and self.tokenizer.bos_token:
+            special_tokens_dict["cls_token"] = self.tokenizer.bos_token
+
+        new_tokens_in_vocab = []
+        for token in [mask_token, bos_token, eos_token, pad_token, sep_token, cls_token, unk_token]:
+            if token is not None and token not in self.tokenizer.get_vocab():
+                new_tokens_in_vocab.append(token)
+
+        if len(new_tokens_in_vocab) > 0:
+            logging.warning(
+                f'{new_tokens_in_vocab} \n will be added to the vocabulary.\n' f'Please resize your model accordingly'
+            )
         self.add_special_tokens(special_tokens_dict)
 
         self.never_split = self.tokenizer.all_special_tokens
-        self.vocab_size = self.tokenizer.vocab_size
+
```

Comment:
What is never_split? does anybody uses that in our examples? If HF tokenizer has such a property, then this property should be synced with that by using property and setter, not to update and maintain by us.

Reply:
Only used inside the tokenizer, removed and replaced with the call to HF tokenizer to get the correct value